src/worker/queue.ts

```diff
@@ -3,23 +3,24 @@ import * as Sentry from '@sentry/node'
 import { DateTime } from 'luxon'
 import Worker from '../celery/worker'
 import Client from '../celery/client'
-import { PluginsServer, Queue, RawEventMessage } from '../types'
+import { PluginsServer, Queue } from '../types'
 import { KafkaQueue } from '../ingestion/kafka-queue'
+import { UUID } from '../utils'
 
-export function startQueue(
+export async function startQueue(
     server: PluginsServer,
     processEvent: (event: PluginEvent) => Promise<PluginEvent | null>,
     processEventBatch: (event: PluginEvent[]) => Promise<(PluginEvent | null)[]>
-): Queue {
+): Promise<Queue> {
     const relevantStartQueue = server.KAFKA_ENABLED ? startQueueKafka : startQueueRedis
-    return relevantStartQueue(server, processEvent, processEventBatch)
+    return await relevantStartQueue(server, processEvent, processEventBatch)
 }
```

Comment:
Awaiting start*Queue to avoid unhandled promise rejections.

Reply:
Actually not awaiting, instead handling start error inside start*Queue with `.catch`.